src/platform-implementation-js/dom-driver/gmail/widgets/gmail-mole-view-driver.ts

```diff
@@ -104,7 +104,24 @@ class GmailMoleViewDriver {
 
   show() {
     const doShow = (moleParent: HTMLElement) => {
-      moleParent.insertBefore(this.#element, last(moleParent.children)!);
+      const lastChild = last(moleParent.children)!;
+
+      if (lastChild instanceof HTMLElement) {
+        /**
+         * 2023-10-02 When Google Chat is enabled, we need to keep track of the `order` style property for moles added.
+         * If we don't, we end up with moles added on top of the sidebar because of a chat element in the same parent element as moleParent.
+         *
+         * This fix ratches the order of the element padding in order to present moles before the sidebar. Whether or not this fix works with Gmail's CSS in a RTL language is unknown.
+         */
+        const oldOrder = lastChild.style.order;
+        const oldOrderNum = parseInt(oldOrder, 10);
+
+        if (!isNaN(oldOrderNum)) {
+          this.#element.style.order = `${oldOrderNum + 1}`;
```

Comment:
When Google Chat is enabled, this approach falls down when 

2. A first Gmail compose view or Chat is opened
3. An SDK mole is opened
4. The first compose view or Chat mole is closed.
5. Another Gmail compose view or Chat is added.

The SDK mole will be on the left most mole instead of on the right. We _could_ potentially listen for compose view events and update order accordingly.

Reply:
The heuristic we'll use here to account for `order` is:

# When Google Chat is enabled

Place new SDK moles on the leftmost edge of any Gmail moles (in LTR layouts). After the first SDK mole is added, the next SDK mole will be added to the left of it.

# When Google Chat isn't enabled

We'll place moles on the rightmost edge of any Gmail moles. After the first SDK mole is added, the next SDK mole will be added to the left of it.

Reply:
Longer term, we're going to try adjusting SDK moles to be in the same DOM Node order as Gmail Compose moles. This will probably mean we have to have a `MutationObserver` on `moleParent`.